ViewEnvironment/Sources/EnvironmentValues+ViewEnvironment.swift

```diff
@@ -29,4 +29,28 @@ extension EnvironmentValues {
     }
 }
 
+
+extension Environment where Value == ViewEnvironment {
+    
+    @available(
+        *,
+         deprecated,
+         message:
+            """
+            Please do not create an `@Environment` property that references the top-level `viewEnvironment`: \
+            it will break SwiftUI's automatic invalidation when any part of the `ViewEnvironment` changes. \
+            Instead, reference your relevant sub-property, eg `@Environment(\\.viewEnvironment.myProperty)`.
+            """
+    )
+    @inlinable public init(_ keyPath: KeyPath<EnvironmentValues, Value>) {
```

Comment:
I can't think of a case where we'd want to do this, but if we want to allow it, perhaps adding an SPI'd `init(unsafe: ...)` would allow that escape hatch.